lib/utils/NodeVisitor.ts

```diff
@@ -3,9 +3,10 @@ import {Base} from '../nodes/Base';
 import {Composite} from '..';
 import {v4} from 'uuid';
 import {ResultCode} from './resultCodes';
+import {IfElse} from '../nodes/IfElse';
 
 function isComposite<S extends BlueshellState, E>(node: Base<S, E>): node is Composite<S, E> {
-	return node instanceof Composite;
+	return node instanceof Composite || node instanceof IfElse;
```

Comment:
Would this be better done by making an interface that defines a children property, have Composite and IfElse implement that interface and making a type guard that this uses instead of isComposite?  You should change the other instances of isComposite() in this file to use that new type guard.  It looks like isComposite() is only used below to make TypeScript aware that the node has a children property.

Reply:
I'm guessing it's something about the way this file iterates over children that is causing the sometimes out-of-order nodes in the rendering...

Reply:
I agree this could be done in a better way.  This is the quick fix, but maybe this predicate should just be renamed to hasChildren and then it'll at least be more accurate to its purpose.

Reply:
That works, yes.

Reply:
Renamed.